Lra 44 setup required fields in tables

britaumich opened the pull request:
This is not a pull request yet because the Infrastructure group hasn't added required fields to the [It Security Incidents] tab in the [Security Tracking Tool - Database fields] spreadsheet. But I want you to review the code.

britaumich committed:
Add validations to legacy_os_record model

britaumich committed:
Add validations to dpa_exception model, edit date fields in new form view

britaumich committed:
Remove time from date fields and make them blank by default in legacy_os_record new form

britaumich committed:
Edit date fields to remove default values and start year from today

britaumich committed:
Add validations to sensitive_data_system model

rsmoke reviewed (commented):
This is very nice. I especially like the custom validator

britaumich committed:
Faled attanpt to add device from legacy os records controller

britaumich committed:
Add logger

britaumich committed:
Add device to legacy os record - validation is not working

britaumich committed:
Commented validation is doing nothing

britaumich committed:
Adding required: true validates serial, but don't keep it's value in form if data type was not selected first

britaumich committed:
Add incomplete checkbox to dpa_exceptions and edit create to save incomplete record

britaumich committed:
Show attachment for different table

britaumich committed:
Add incomplete checkbox to it_security_incidents and edit create to save incomplete record

britaumich committed:
Fix stupid error to make a select field required

britaumich committed:
Add attachments in show view

britaumich committed:
Restore originals

britaumich committed:
Add attachments in show view

britaumich committed:
Fix validation for mandatory and required fields for it_security_incidents

britaumich committed:
Add validation for mandatory and required fields for legacy_os_records

britaumich committed:
Add validation for mandatory and required fields for dpa_exceptions

britaumich committed:
Add validation for mandatory and required fields for sensitive_data_systems

britaumich committed:
Fix comflict

britaumich committed:
Improve create and api methods to search TDX database by serial and hostname and check results for only one return

britaumich commented:
I merged LRA-42-device-creation-for-main-tables to this branch because I had to change the database tables (and hence views) in this branch and therefore couldn't work in LRA-42-device-creation-for-main-tables branch (tables and view didn't match).
The following I added to the device-create function:
- search the TDX assets database by serial number and hostname (user enter a serial or hostname into [Add device] field)
- error message in case multiple records are returned by TDX API
- create a device with hostname or serial

britaumich committed:
Add hostname to form and API search

britaumich committed:
Add validation for either serial or hostname

britaumich committed:
Move TDX API into a module

britaumich committed:
Make device an instance variable; make method independant  of models

britaumich committed:
Use DeviceTdxApi module instead of build in method

britaumich committed:
Make script model independant

britaumich committed:
Create device before calling DeviceTdxApi

britaumich committed:
Change view to use DeviceTDxApi for the rest of devoce data

britaumich committed:
Add validation - it's not working as expected

britaumich committed:
Add edit method with API call

britaumich committed:
Add incomlete scope for devices and list of incomplete devices to dashboard

britaumich committed:
Remove unnesasary code

britaumich committed:
Add incomplete check - not sure if it's correct; the scope too

britaumich committed:
Add show which calls device_tdx_api if devose is incomplete

britaumich commented:
I improved the TDX API method (get_device_data) a lot:
- moved it to a module
- call it from controllers: legacy_os_record create and show and device create methods
Added incomplete scope and incomplete? method to the device model  (not sure if queries are correct)

britaumich committed:
Fix typo

britaumich committed:
Create model test for mandatory and required fields for legacy_os_record model

britaumich committed:
Create model test for mandatory and required fields for dpa_exception model

britaumich committed:
Commented validation because it works incorrectly for legacy_os_record model

rsmoke reviewed (commented):
This is pretty
